sched/scheduler/stateful_scheduler.go

```diff
@@ -92,11 +105,11 @@ func NewStatefulScheduler(
 
 	// TODO: we need to allow the scheduler to accept new jobs
 	// while recovering old ones.
-	if recoverJobsOnStartup {
+	if config.RecoverJobsOnStartup {
 		sched.startUp()
 	}
 
-	if !debugMode {
+	if !config.DebugMode {
```

Comment:
in case you want to bring scheduler up in Debug mode, otherwise you'd have this weird race.  of bringing it up and then having to send an API command.